### PR TITLE
Bitbucket connector

### PR DESCRIPTION
**Pull request by chandru094**

### CODE REVIEW

**Comment by huilinn** [comment]
Auth doesn't seem to be configured properly:
screen shot 2016-08-13 at 12 14 52 am

**Comment by chandru094** [comment]
Hi

Can you please check if the callback URL for the consumer you created in
Butbucket is correct.

Regards,
Chandra Prakash R

**Comment by huilinn** [comment]
Oh, @chandru094 I think I was using your localhost client ID and secret. When I replaced it with your production ID and secret in the code, it worked.

However, I connected immediately to my browser's Bitbucket account with no regards to what I type in for the input fields - does this mean we don't actually need the input fields to be there?
[https://gyazo.com/f276e5e1cff9c6fc704838d126b6071c](https://gyazo.com/f276e5e1cff9c6fc704838d126b6071c)

**Comment by chandru094** [comment]
The username and repository name are required in all the endpoints url. So
I added them in the connection.

**Comment by huilinn** [comment]
Ah okay I see. By the way @chandru094 can you change the client id and secret in the PR? Right now they're still configured to your localhost creds. Production ones:

       client_id: 'HV8cTJ8aLyKC8jucUh',
       client_secret: 'BGHemmMqvXTkPAdCryzTw5GWcw82M8Vx',

**Comment by chandru094** [comment]
I have changed the client id and client secret now to be used by production accounts.

**Comment by huilinn** [comment]
Actually, @chandru094, it looks like what's required to be passed into every API call is not username and repo slug, but owner's username and repo slug. So the hint should be changed for the connection to say that the username passed in must be that of the owner of the repo.

Was looking at another API call:
screen shot 2016-08-14 at 1 37 59 am

**Comment by huilinn** [comment]
Also, I think we're currently not handling the case for repos with >1 word. The slug would be all lowercase, with - replace spaces. For example, my site's repo name and slug name:
screen shot 2016-08-14 at 1 58 09 am

The API calls we're making simply passes the user's entered repo name into the call currently, we should change that.

**Comment by huilinn** [comment]
Instead of santhosh requesting permission, we should have Workato requesting permission, for a global scope app: screen shot 2016-08-14 at 2 06 05 am

**Comment by huilinn** [comment]
Also, for priority and kinds fields, it would be better if they were freeform text fields instead of picklists (we can also have them defaulted to text fields, and then toggle to picklists), because we want that flexibility to assign different priorities and kinds when creating an issue. If it was a picklist, we would be hardcoding all issues to be a specific priority and specific kind, and that's not good for a recipe.

Generally, most fields shouldn't be a picklist field (i.e. issue state, issue type, issue priority). 

screen shot 2016-08-14 at 2 13 57 am

**Comment by chandru094** [comment]
If we pass any other values other than values given in the api it throws an error saying not a valid value.

**Comment by eeshan92** [comment]
@chandru094 @huilinn toggle feature will fix that problem. In the meantime, let's change it to a free-text field for the use case

File: oauth2/bitbucket_connector.rb [hunk 1]
```diff
@@ -0,0 +1,266 @@
+{
+  title: 'bitbucket',
```
Comment (eeshan92) [inline oauth2/bitbucket_connector.rb]: Titleize this to `Bitbucket`

File: oauth2/bitbucket_connector.rb [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+      token_url: ->() {
+       'https://bitbucket.org/site/oauth2/access_token?type=refresh'
+      },
+
+      client_id: 'HV8cTJ8aLyKC8jucUh',
+
+      client_secret: 'BGHemmMqvXTkPAdCryzTw5GWcw82M8Vx',
+
+      credentials: ->(connection, access_token) {
+      headers('Authorization': "Bearer #{access_token}")
```
Comment (eeshan92) [inline oauth2/bitbucket_connector.rb]: should be indented

**Pull request closed without merging**